sched/sched/sched_suspendscheduler.c

```diff
@@ -63,7 +63,7 @@ void nxsched_suspend_scheduler(FAR struct tcb_s *tcb)
 {
   /* Handle the task exiting case */
 
-  if (tcb != NULL)
+  if (tcb == NULL)
```

Comment:
It is impossible for the caller of this function to pass an invalid tcb, right? let us remove this check.

Reply:
pthread_terminate/pthread_exit/task_exit may pass NULL tcb, so it's better to do the check here instead spreading in many place.

Reply:
In the latest implementation, we will call nxsched_suspend_scheduler(g_running_tasks[this_cpu()]); and g_running_tasks[this_cpu()] will be set to NULL in up_exit

Reply:
`nxsched_resume_scheduler()` and `nxsched_suspend_scheduler()` are paired functions. If the code resumes but skips suspend, the format in the trace will be wrong, which is problematic for backend IDEs such as system view/perfetto.

Reply:
@XuNeo cc

Reply:
@anchao this patch only fix the regression on !tcb -> tcb == NULL. we will try to make mempool compatible with relay free to allow when exit, reuse tcb after exit. we should keep the nxsched_suspend_scheduler no regression at higher priority.

Reply:
I prefer to remove the check here，and move the relative logic into arch exit, which hard to implement?

Reply:
Next patch, we will modify the exit free logic, and remove the "if" check here.

Reply:
@anchao